File: IDE/temp files/newfile243366.js


function LoopParseFunc(varString, delimiter1="", delimiter2="") {
    let items;
    if (!delimiter1 && !delimiter2) {
        // If no delimiters are provided, return an array of characters
        items = [...varString];
    } else {
        // Construct the regular expression pattern for splitting the string
        let pattern = new RegExp('[' + delimiter1.replace(/[.*+?^${}()|[\]\\]/g, '\\$&') + delimiter2.replace(/[.*+?^${}()|[\]\\]/g, '\\$&') + ']+');
        // Split the string using the constructed pattern
        items = varString.split(pattern);
    }
    return items;
}

function print(value) {
    console.log(value)
}

function StrLen(s) {
    // Return the length of the given string
    return s.length;
}

function SubStr(str, startPos, length) {
    // If str is null or undefined, return an empty string
    if (str === null || str === undefined) {
        return "";
    }
    // If length is not provided or is blank, default to "all characters"
    if (length === undefined || length === "") {
        length = str.length - startPos + 1;
    }
    // If startPos is less than 1, adjust it to start from the end of the string
    if (startPos < 1) {
        startPos = str.length + startPos;
    }
    // Extract the substring based on startPos and length
    return str.substr(startPos - 1, length);
}


async function SubStrLastChars(text, numOfChars) {
    var strLength = StrLen(text);
    if (numOfChars > strLength) {
        return text;
    }
    let textArr = [];
    items1 = LoopParseFunc(text)
    for (let A_Index1 = 0; A_Index1 < items1.length + 0; A_Index1++) {
        const A_LoopField1 = items1[A_Index1 - 0];
        textArr.push(A_LoopField1);
    }
    var LastOut = "";
    for (let A_Index2 = 0; A_Index2 < numOfChars + 0; A_Index2++) {
        LastOut = textArr[(strLength - A_Index2)] + LastOut;
    }
    return LastOut;
}
async function main() {
    print(await SubStrLastChars("hello", 2));
}
main();

File: IDE/temp files/newfile243366.py
import re

def LoopParseFunc(var, delimiter1="", delimiter2=""):
    if not delimiter1 and not delimiter2:
        # If no delimiters are provided, return a list of characters
        items = list(var)
    else:
        # Construct the regular expression pattern for splitting the string
        pattern = r'[' + re.escape(delimiter1) + re.escape(delimiter2) + r']+'
        # Split the string using the constructed pattern
        items = re.split(pattern, var)
    return items

# used the print func

def StrLen(s: str) -> int:
    # Return the length of the given string
    return len(s)

def SubStr(s, startPos, length=None):
    if s is None or s == "":
        return ""
    if length is None or length == "":
        length = len(s) - startPos + 1
    if startPos < 1:
        startPos = len(s) + startPos
    return s[startPos - 1:startPos - 1 + length]


def SubStrLastChars(text, numOfChars):
    strLength = StrLen(text)
    if (numOfChars > strLength):
        return text
    textArr = []
    items1 = LoopParseFunc(text)
    for A_Index1 , A_LoopField1 in enumerate(items1, start=0):
        textArr.append(A_LoopField1)
    LastOut = ""
    for A_Index2 in range(0, numOfChars + 0):
        LastOut = textArr[(strLength - A_Index2)] + LastOut
    return LastOut
print(SubStrLastChars("hello", 2))